openbas-api/src/main/java/io/openbas/service/EndpointService.java

```diff
@@ -142,6 +142,37 @@ public Page<Endpoint> searchEndpoints(SearchPaginationInput searchPaginationInpu
         Endpoint.class);
   }
 
+  public Page<Endpoint> searchManagedEndpoints(
+      Specification<Endpoint> spec,
+      String assetGroupId,
+      SearchPaginationInput searchPaginationInput) {
+    AssetGroup assetGroup = assetGroupRepository.findById(assetGroupId).get();
```

Comment:
this.assetGroupRepository
            .findById(assetGroupId)
            .orElseThrow(() -> new IllegalArgumentException("Asset group not found"));